### PR TITLE
Improve error handling of `gdel` & `gforw`

### PR DESCRIPTION
**Pull request by glenshelly**
Updated the git functions `gdel` and `gforw` to add
error handling and usage patterns

### CODE REVIEW

File: git/functions/gforw [hunk 1]
```diff
@@ -1,5 +1,19 @@
...
+  echo ''
+  echo 'USAGE:'
+  echo ''
+  echo '   gforw NUMBER-OF-COMMITS [BRANCH-NAME]'
+  echo ''
+  echo '...where BRANCH-NAME is the branch towards which to move (assumes ''master'' if not given)'
+  echo ''
+  return 1
+}
+
```
Comment (nathanshelly) [inline git/functions/gforw]: same Here doc suggestion as above, would look like:

```bash
[[ -z $1 ]] && {
  cat << EOF
error: the following arguments were not provided

  NUMBER-OF-COMMITS

USAGE:

  gforw NUMBER-OF-COMMITS [BRANCH-NAME]

...where BRANCH-NAME is the branch towards which to move (assumes master if none given)
EOF
  return 1
}
```

File: git/functions/gdel [hunk 2]
```diff
@@ -3,7 +3,14 @@
...
 [[ -z $1 ]] && {
-  echo 'No branch given, aborting'
+  echo 'error: the following arguments were not provided'
+  echo ''
+  echo '  BRANCH-NAME'
+  echo ''
+  echo 'USAGE:'
+  echo ''
+  echo '    gdel BRANCH-NAME'
+  echo ''
```
Comment (nathanshelly) [inline git/functions/gdel]: [Here docs](https://en.wikipedia.org/wiki/Here_document#Unix_shells) can make this a little nicer. Would like below:

> Note: spacing is a little weird inside Here docs, all whitespace/indentation choices below are purposeful

```bash
[[ -z $1 ]] && {
  cat << EOF
error: the following arguments were not provided

  BRANCH-NAME

USAGE:

  gdel BRANCH-NAME
EOF
  return 1
}
```

File: git/functions/gdel [hunk 3]
```diff
@@ -3,7 +3,14 @@
...
 
 [[ -z $1 ]] && {
-  echo 'No branch given, aborting'
+  echo 'error: the following arguments were not provided'
+  echo ''
+  echo '  BRANCH-NAME'
+  echo ''
+  echo 'USAGE:'
+  echo ''
+  echo '    gdel BRANCH-NAME'
```
Comment (nathanshelly) [inline git/functions/gdel]: to make it even more accurate could add `...` to indicate that multiple `BRANCH-NAME`s can be given - `  gdel BRANCH-NAME...`

File: git/functions/gforw (hunk 1 above)
Comment (glenshelly) [reply git/functions/gforw]: done

File: git/functions/gdel (hunk 3 above)
Comment (glenshelly) [reply git/functions/gdel]: done

**Pull request merged**